src/editor/Collaboration.js

```diff
@@ -0,0 +1,70 @@
+define(function (require, exports, module) {
+    "use strict";
+
+    var SimpleWebRTC = require("simplewebrtc");
+
+    function Collaboration() {
+        var self = this;
+        var webrtc = new SimpleWebRTC({
+          // the id/element dom element that will hold "our" videos
+          localVideoEl: 'localVideo',
+          // the id/element dom element that will hold remote videos
+          remoteVideosEl: 'remotesVideos',
+          // immediately ask for camera access
+          autoRequestMedia: true
+        });
+        this.webrtc = webrtc;
+        this.webrtc.on('readyToCall', function () {
+          webrtc.joinRoom('thimble');
+        });
+        this.init();
+    };
+
+    Collaboration.prototype.init = function() {
+        var self = this;
+        self.changing  = false;
```

Comment:
I'm curious, is this blocking logic around `self.changing` really necessary?  Are there actually times where you'd have multiple events happening at once?  It seems unnecessary to me, in a single threaded DOM, but maybe I'm missing something?  All your code below is synchronous, so it shouldn't cause issues.